lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -799,21 +824,31 @@ def generate_docker_py_service_description(self, name, docker_networks):
                 )
             )
 
+        dns_config = types.DNSConfig(
+            nameservers=self.dns,
```

Comment:
I think it would help if you could move all these "cleanup" parts of this PR which aren't part of the feature into a separate PR. That allows to backport that PR to stable-2.7; if these changes are part of a New Feature PR, they cannot be backported that easily (feature backports are prohibited in general) and that could make it harder to backport later bugfixes which touch something in the refactored parts.

Reply:
Fixing such stuff is in general a good idea, but it's better to do it in a PR which can be backported ;-)